crates/llama-cpp-bindings/src/engine.cc

```diff
@@ -280,6 +307,11 @@ class TextInferenceEngineImpl : public TextInferenceEngine {
 
         int32_t i_batch = request.i_batch - i;
         float* logits = llama_get_logits_ith(ctx, i_batch);
+            // TODO: Replace this logic with nmax 5 to get indices
+            // then construct an array of logits (floats) from indices
+            // then run softmax on that array
+            // then run weighted_random on that array to get an index
+            // note: we might not actually need nmax since we'll need to check the predicate on each index (logit) first
```

Comment:
let's get this implemented with top-5 to update the golden test as a first step.